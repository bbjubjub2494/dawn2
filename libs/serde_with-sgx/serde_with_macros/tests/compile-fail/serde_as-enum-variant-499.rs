use serde::{Deserialize, Serialize};
use serde_with_macros::serde_as;

#[serde_as]
#[derive(Serialize, Deserialize)]
#[serde(tag = "messageType", content = "content")]
/// The contents of a message combined with the `MessageType`
pub enum MessageContentJsonStringEnum {
    /// A normal message
    Text(String),
    /// Fancy object message
    #[serde_as(as = "serde_with::json::JsonString")]
    Object(String),
}

fn main() {}
