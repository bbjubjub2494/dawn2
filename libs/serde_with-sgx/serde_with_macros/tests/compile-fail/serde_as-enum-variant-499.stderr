error: serde_as attribute is not allowed on enum variants
  --> tests/compile-fail/serde_as-enum-variant-499.rs:12:5
   |
12 |     #[serde_as(as = "serde_with::json::JsonString")]
   |     ^
